docs/DafnyRef/Topics.md

```diff
@@ -1158,12 +1158,16 @@ There are particular expressions which, for technical reasons, Dafny can not use
 
 Besides not finding triggers, another problematic situation is when Dafny was able to generate triggers, but believes the triggers it found may still cause useless instantiations because they create matching loops. Dafny emits a warning when this happens, indicating the Dafny program should be changed so Dafny can find triggers for this quantifier that do not cause matching loops.
 
-To understand matching loops, one needs to understand how triggers are used. During verification, Dafny maintains a set of expressions which it believes to be true, which we call ground terms. Given a univeral quantifier that's a ground term, Dafny will use its triggers to pattern match on other ground terms. If the pattern matches, that ground term is used to instantiate the quantifier. 
+To understand matching loops, one needs to understand how triggers are used. During a single verification run, such as verifying a method or function, Dafny maintains a set of expressions which it believes to be true, which we call the ground terms. For example, in the body of a method, Dafny knows the requires clauses of that method hold, so the expressions in those will be ground terms. When Dafny steps through the statements of the body, the set of ground terms grows. For example, when an assignment `var x := 3` is evaluated, a ground term `x == 3` will be added. Given a universal quantifier that's a ground term, Dafny will try to pattern match its triggers on other ground terms. If the pattern matches, that ground term is used to instantiate the quantifier. 
```

Comment:
I think if we introduce a new word we should ensure it is used often enough. Knowledge base is used just once. Regarding 'trigger' and 'ground terms', these concepts come from SMT. Even though Dafny encapsulates SMT, there are some holes in that encapsulation so I think reusing some of those terms makes sense at this point.

Reply:
Oh ok. I'm fine with ground terms then. I'll just have a minor suggestion then, because pattern matching is done not just at the level of ground terms, but sub-expressions
```suggestion
To understand matching loops, one needs to understand how triggers are used. During a single verification run, such as verifying a method or function, Dafny maintains a set of expressions which it believes to be true, which we call the ground terms. For example, in the body of a method, Dafny knows the requires clauses of that method hold, so the expressions in those will be ground terms. When Dafny steps through the statements of the body, the set of ground terms grows. For example, when an assignment `var x := 3` is evaluated, a ground term `x == 3` will be added. Given a universal quantifier that's a ground term, Dafny will try to pattern match its triggers on sub-expressions of other ground terms. If the pattern matches, that sub-expression is used to instantiate the quantifier. 
```